### PR TITLE
eviction: add global histogram for iteration durations

### PR DESCRIPTION
**Pull request by problame**
I would like to know whether and by how much the eviction iterations spike in the $period-sized window that happens every $threshold , when all the timelines do the imitate accesses.

refs https://github.com/neondatabase/neon/issues/4154

### CODE REVIEW

File: pageserver/src/metrics.rs [hunk 1]
```diff
@@ -186,6 +186,16 @@ static PERSISTENT_BYTES_WRITTEN: Lazy<IntCounterVec> = Lazy::new(|| {
     .expect("failed to define a metric")
 });
 
+pub(crate) static EVICTION_ITERATION_DURATION: Lazy<HistogramVec> = Lazy::new(|| {
+    register_histogram_vec!(
+        "pageserver_eviction_iteration_duration_seconds_global",
+        "Time spent on a single eviction iteration",
+        &["period_secs", "threshold_secs"],
+        STORAGE_OP_BUCKETS.into(), // TODO: could use better buckets?
```
Comment (koivunej) [inline pageserver/src/metrics.rs:194]: All of last week fits in 16.7min with (on the latest 6h with 16.7min entries):

- <10ms: 23k
- <100ms: 16k
- <1s: 17k
- <10s: 53
- <100s: 26
- <1000s: 3

I guess we could fit however multimodal distribution there. But trying to guess better buckets, I don't know how much wiser could we become. Capturing non-bucketed data would be much better, but I don't think prometheus fits into that picture.

So suggesting:

```suggest
        STORAGE_OP_BUCKETS.into(),
```

I don't know why can't I suggest.

File: pageserver/src/tenant/timeline/eviction_task.rs [hunk 2]
```diff
@@ -120,6 +120,13 @@ impl Timeline {
                 }
                 let elapsed = start.elapsed();
                 crate::tenant::tasks::warn_when_period_overrun(elapsed, p.period, "eviction");
+                crate::metrics::EVICTION_ITERATION_DURATION
+                    .get_metric_with_label_values(&[
+                        &format!("{}", p.period.as_secs()),
+                        &format!("{}", p.threshold.as_secs()),
```
Comment (koivunej) [inline pageserver/src/tenant/timeline/eviction_task.rs:126]: Unfortunate to-stringing... But perhaps does not matter enough to hoist this to upper level and do the tricks which already panicked in staging again.

**Review by koivunej — APPROVED** [review]
Would remove the TODO comment, ignore my note about to-stringing.

File: pageserver/src/tenant/timeline/eviction_task.rs (hunk 2 above)
Comment (problame) [reply pageserver/src/tenant/timeline/eviction_task.rs:126]: Yeah. I didn't want to go through the trouble of building the equivalent of `EvictionsWithLowResidenceDurationBuilder` for this metric.

File: pageserver/src/metrics.rs (hunk 1 above)
Comment (problame) [reply pageserver/src/metrics.rs:194]: Not sure how they relate to this one.
I guess gc and compaction are somewhat comparable in the sense that they're also background jobs and also do some getpage requests

Comment (koivunej) [reply pageserver/src/metrics.rs:194]: Added that source for the values were for logical size calculation. Cached values do not count in these values.

Comment (problame) [reply pageserver/src/metrics.rs:194]: Ok, then I guess these are good enough for starters.

**Pull request merged**